twitter_user returns empty for user names with underscore after first letter

**Describe the bug**
If I query the twitter_user table for a user name that has an underscore after the first letter I get an empty response. 

**Steampipe version (`steampipe -v`)**
Example: v0.16.4

**Plugin version (`steampipe plugin list`)**
Example: v0.2.0

**To reproduce**
Verify API return data with curl:

```
curl --request GET 'https://api.twitter.com/2/users/by/username/a_garfield101' --header 'Authorization: Bearer ....'

{"data":{"id":"108656083","name":"Andrew Garfield","username":"a_garfield101"}}
```

Check steampipe output

```
> .output line
> select * from twitter_user where username='a_garfield101'
>

```

Try a user with leading underscore to show in some cases it works

```
> select * from twitter_user where username='_fel1x'
-[ RECORD 1  ]---------------------------------------------------------------------------
id                | 212962007
name              | Felix Wilhelm
username          | _fel1x
created_at        | 2010-11-07T10:27:05-05:00
description       | Cloud security research at Project Zero
entities          | <null>
location          | Zürich, Switzerland
<snip>
_ctx              | {"connection_name":"twitter"}
```

**Expected behavior**
Information returned from the API about the user is displayed. 

**Additional context**

Tested on Mac Monterey 12.6 

I'm fairly new to using steampipe so maybe the problem is between the chair and keyboard.

Thanks @Rich5 for spotting the bug and adding so many details to the issue 👍.

I was able to successfully replicate this issue and you were right, the underscore definitely doesn't matter.

I ran the above query with the logger and had the following observations:

```
2022-10-25 16:57:28.980 UTC [ERROR] steampipe-plugin-twitter.plugin: [ERROR] 1666717048887: twitter_user.listUser: query_error="&{Not Found Error Could not find tweet with pinned_tweet_id: [935639732112982017]. https://api.twitter.com/2/problems/resource-not-found tweet pinned_tweet_id 935639732112982017}" opts="{[pinned_tweet_id] [id text attachments author_id context_annotations conversation_id created_at entities geo in_reply_to_user_id lang public_metrics possibly_sensitive referenced_tweets source withheld] [created_at description entities id location name pinned_tweet_id profile_image_url protected public_metrics url username verified withheld]}"
```

The issue occurs for the `pinned_tweet` column since the pinned_tweet_id (935639732112982017) doesn't exist.

Here is a sample JSON response that I managed to grab from the Twitter API playground - 
```
{
  "data": {
    "pinned_tweet_id": "935639732112982017",
    "username": "a_garfield101",
    "name": "Andrew Garfield",
    "id": "108656083"
  },
  "errors": [
    {
      "value": "935639732112982017",
      "detail": "Could not find tweet with pinned_tweet_id: [935639732112982017].",
      "title": "Not Found Error",
      "resource_type": "tweet",
      "parameter": "pinned_tweet_id",
      "resource_id": "935639732112982017",
      "type": "https://api.twitter.com/2/problems/resource-not-found"
    }
  ]
}
```

I am looking at a possible fix wherein we would return `nil` for the `pinned_tweet` column whenever the `pinned_tweet_id` column returns a `Not found Error` code.

--- a/twitter/table_twitter_user.go
+++ b/twitter/table_twitter_user.go
@@ -51,7 +51,7 @@ func listUser(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (
 		return nil, lookupErr
 	}
 
-	softError := []*twitter.ErrorObj{}
+	softErrors := []*twitter.ErrorObj{}
 
 	// Soft error, e.g. 404
 	if len(result.Raw.Errors) > 0 {
@@ -65,13 +65,13 @@ func listUser(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (
 				}
 				return nil, nil
 			}
-			softError = append(softError, e)
+			softErrors = append(softErrors, e)
 		}
 	}
 
-	if len(softError) > 0 {
+	if len(softErrors) > 0 {
 		errMsgs := []string{}
-		for _, e := range softError {
+		for _, e := range softErrors {
 			errMsgs = append(errMsgs, fmt.Sprintf("%s: %s", e.Title, e.Detail))
 		}
 		// Return the full set of error messages

--- a/twitter/table_twitter_user.go
+++ b/twitter/table_twitter_user.go
@@ -55,10 +55,8 @@ func listUser(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (
 
 	// Soft error, e.g. 404
 	if len(result.Raw.Errors) > 0 {
-		errMsgs := []string{}
 		for _, e := range result.Raw.Errors {
 			plugin.Logger(ctx).Error("twitter_user.listUser", "query_error", e, "opts", opts)
-			errMsgs = append(errMsgs, fmt.Sprintf("%s: %s", e.Title, e.Detail))
 			
 			// Check if the Not Found Error is a result of "pinned_tweet_id" parameter, return nil
 			if e.Title == "Not Found Error" {

--- a/twitter/table_twitter_user.go
+++ b/twitter/table_twitter_user.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"strings"
 
+
 	twitter "github.com/g8rswimmer/go-twitter/v2"
 
 	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
@@ -49,19 +50,36 @@ func listUser(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (
 		plugin.Logger(ctx).Error("twitter_user.listUser", "query_error", lookupErr, "opts", opts)
 		return nil, lookupErr
 	}
+
+	softError := []*twitter.ErrorObj{}
+
 	// Soft error, e.g. 404
 	if len(result.Raw.Errors) > 0 {
 		errMsgs := []string{}
 		for _, e := range result.Raw.Errors {
 			plugin.Logger(ctx).Error("twitter_user.listUser", "query_error", e, "opts", opts)
 			errMsgs = append(errMsgs, fmt.Sprintf("%s: %s", e.Title, e.Detail))
+			
+			// Check if the Not Found Error is a result of "pinned_tweet_id" parameter, return nil
 			if e.Title == "Not Found Error" {
+				if e.Parameter == "pinned_tweet_id" {
+					continue
+				}
 				return nil, nil
 			}
+			softError = append(softError, e)
+		}
+	}
+
+	if len(softError) > 0 {
+		errMsgs := []string{}
+		for _, e := range softError {
+			errMsgs = append(errMsgs, fmt.Sprintf("%s: %s", e.Title, e.Detail))
 		}
 		// Return the full set of error messages
 		return nil, errors.New(strings.Join(errMsgs, "\n"))
 	}
+
 	for _, i := range result.Raw.UserDictionaries() {
 		d.StreamListItem(ctx, i)
 	}

